Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,173 @@
+
+In the optimistic way, once the oracle signature is released, parties interact
+off-chain to settle the DLC. They generate a *closing_tranaction* and publish it on the
+blockchain.
+
+In the pessimitic way, a party interact with the chain to unilaterally settle the DLC.
+They are two scenarios to consider:
+
+1.  The oracle releases the outcome signature but the DLC counterparty goes wrong,
+either going offline or maliciously behaving. The party must broadcast the DLC
+transaction paired with the outcome.
+
+2. The oracle doesn't release the outcome signature, independently of counterparty
+behavior, the party must broadcast the refund transaction after its timelock expiration.
+
+# Table of Contents
+	* [General Requirements](#general-requirements)
+	* [Funding Phase](#funding-phase)
+	* [Execution Phase](#execution-phase)
+	* [Fee-Bumping Policy](#fee-bumping-policy)
+	* [Reorgs] (#reorgs)
+
+# General Requirements
+
+A DLC client should have censorship-resistant access to the blockchain, use a
+local fee-estimator and ensure it's well-connected to the tx-relay network.
+
+We recommend to not react on any mempool event as otherwise it would be a source of
+cross-layer mapping.
+
+We recommend to not sign local transactions until their broadcast is required
+to avoid their misuage in case of site compromise, assuming an external signer.
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transaction to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+	- if a counterparty's funding signature is received:
+		- MUST finalize the funding transaction with a local `SIGHASH_ALL` signature
+		- MUST broadcast the funding transaction
+
+A node:
+	- if a funding signature has been sent or a funding transaction broadcast and there is no confirmation after 6 blocks:
+		- MUST spend the change output with a CPFP increasing package feerate to confirm soonly
+		- MUST start a bumping timer of length `congestion_delay`
+		- if `congestion_delay` expires:
+			- MUST spend the change output with a CPFP increasing package feerate to confirm soonly
+			- MUST replace according to BIP 125
+			- MUST reschedule timer for next bump
+		- if `contract_maturity_bound` is 20 blocks ahead from local tip:
+			- MUST spend the funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST start a bumping timer of length `security_delay`
+			- MUST disable `congestion_delay` timer
+		- if `security_delay` expires:
+			- MUST spend the funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST replace according to BIP 125
+			- MUST reschedule timer for next bump
+		- MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+	- if the funding transaction confirms:
+		* MUST track spend of the CET output
```

Comment:
Not sure I get why you need to track spend of the CET output, do you mean the fund tx output?

Reply:
If the funding transaction confirms and at CET maturing your counterparty broadcast and gets confirmed the CET you must detect it and mark for spend your balance output.

You can parse onchain CETs by their TXIDs by if you have a thousand of them that's more efficient to track the funding output spent.

Note, you want your onchain parser to be efficient if protocol participants are delegating their CET monitoring to some third-party entity like a watchtower which may have tens of thousands of CET to watch.